src/tools/isabelle_executor.py

```diff
@@ -611,6 +609,25 @@ def _handle_auto_tactics(self, start_state: str, end_state: str) -> str:
             raise Exception(description)
         return description.split('<hammer>')[0] + '<hammer>'
 
+    def _handle_reg_tactic(self, start_state: str, step: str, end_state: str, proof_search_mode=True) -> str:
+        description = self.pisa_env.step(start_state, step, end_state)
+        if not description.startswith('Step error:'):
+            return step
+        
+        tactics = re.split(r'\susing\s|\sby\s', step, maxsplit=1)
```

Comment:
make the assumption that tactics are of the form
`have "goal" (using x) by <tac>`
This is not always true, but is ok